core/domain/classifier_services.py

```diff
@@ -83,8 +83,13 @@ def classify_string_classifier_rule(state, normalized_answer):
     best_matched_answer_group_index = len(state.interaction.answer_groups)
     best_matched_rule_spec_index = None
 
-    sc = classifier_registry.Registry.get_classifier_by_id(
-        feconf.INTERACTION_CLASSIFIER_MAPPING['TextInput'])
+    if state.classifier_model_id == None:
+        sc = classifier_registry.Registry.get_classifier_by_id(
```

Comment:
It seems to me like the "id" here refers to two different things in the different branches. In one case it refers to the algorithm id and in the other case it refers to the state's classifier model id. So there seems to be some confusion.

Can we perhaps do a separate PR first to rename the Registry method to get_classifier_by_algorithm_type or similar? /cc @anmolshkl